core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -216,14 +233,73 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
     } yield clone
 
   def cloneAzureWorkspace(sourceWorkspace: Workspace,
-                          project: RawlsBillingProject,
                           profile: ProfileModel,
-                          request: WorkspaceRequest
-  ): Future[Workspace] = Future.failed(
-    RawlsExceptionWithErrorReport(
-      ErrorReport(StatusCodes.NotImplemented, "WOR-625")
-    )
-  )
+                          request: WorkspaceRequest,
+                          parentContext: RawlsRequestContext
+  ): Future[Workspace] = {
+
+    def createNewWorkspaceRecord(): Future[Workspace] =
+      dataSource.inTransaction { access =>
+        for {
+          _ <- failIfWorkspaceExists(request.toWorkspaceName)
+          workspaceId = UUID.randomUUID()
+          newWorkspace <- createMultiCloudWorkspaceInDatabase(
+            workspaceId.toString,
+            request.toWorkspaceName,
+            request.attributes,
+            access,
+            parentContext
+          )
+        } yield newWorkspace
+      }
+
+    for {
+      // The call to WSM is asynchronous. Before we fire it off, allocate a new workspace record
+      // to avoid naming conflicts - we'll erase it should the clone request to WSM fail.
+      newWorkspace <- createNewWorkspaceRecord()
+
+      cloneResult <- traceWithParent("workspaceManagerDAO.cloneWorkspace", parentContext) { context =>
+        Future(blocking {
+          workspaceManagerDAO.cloneWorkspace(
+            sourceWorkspace.workspaceIdAsUUID,
+            newWorkspace.workspaceIdAsUUID,
+            request.name,
+            profile.getId,
+            request.bucketLocation.getOrElse(""),
```

Comment:
thanks

Reply:
Christina tells me that this location is determined by the landing zone's storage account. I'm now in a bit of a knot as to how to support this in a generic way as I understand this route is meant to work for both GCP and azure workspaces.

Reply:
Maybe we won't have to worry about this as we transition from rawls, but if we were, I might consider refactoring the `ProfileModel` to be a tagged-union instead of an all-in-one jobby. Likewise workspaces and billing projects.

Reply:
Terra-ui currently _does_ send a location when creating or cloning an Azure workspace because it sends a default value (on create we just ignore it for an Azure workspace). But I don't think it should for Azure workspaces because we don't allow the user to choose a location for those workspaces… I can update my PR to stop sending `bucketLocation` if the workspace's billing project is Azure.

This is the current paylod  for an Azure workspace:

````
attributes: {description: ""}
authorizationDomain: []
bucketLocation: "US-CENTRAL1"
copyFilesWithPrefix: "notebooks/"
name: "CARAzureDec13 copy"
namespace: "CARDec13Work"

Reply:
That would be why you were hitting that error because I had previously enforced that if you specified a location, it was whatever our default value was.
I think we should ignore the bucketLocation property. If we ever allow users to create storage containers in locations other than that determined by their LZ, we should add a distinct field for this.